mono/metadata/method-builder-ilgen.c

```diff
@@ -544,10 +544,16 @@ mono_mb_emit_native_call (MonoMethodBuilder *mb, MonoMethodSignature *sig, gpoin
 }
 
 void
-mono_mb_emit_icall (MonoMethodBuilder *mb, gpointer func)
+mono_mb_emit_icall_id (MonoMethodBuilder *mb, MonoJitICallId jit_icall_id)
 {
```

Comment:
The old function name can stay, it would reduce the diff.

Reply:
I'll try that. Elsewhere we need a macro with the old name to reduce diff, and occasional use of function for less common but still recursing cases. I'll check.

Reply:
Maybe but I don't think so.
I kept the name as a macro for the common case -- when there is a constant identifier and not a variable, but then some cases cannot use that, and call the new function.
The ratio is about 25%,.

jay@jaykrell-tp2:/mnt/c/s/mono2/mono$ grep  mono_mb_emit_icall_id */*.{c,h} |wc -l
28
jay@jaykrell-tp2:/mnt/c/s/mono2/mono$ grep  mono_mb_emit_icall\  */*.{c,h} |wc -l
98

Reply:
Makes sense.